redfish/provider/data_source_redfish_virtual_media.go

```diff
@@ -1,110 +1,136 @@
-// 						Computed:    true,
-// 					},
-// 					"id": {
-// 						Type:        schema.TypeString,
-// 						Description: "Id of the virtual media resource",
-// 						Computed:    true,
-// 					},
-// 				},
-// 			},
-// 		},
-// 	}
-// }
+// DellVirtualMediaDatasource is new datasource for group devices
+func NewDellVirtualMediaDatasource() datasource.DataSource {
+	return &DellVirtualMediaDatasource{}
+}
 
-// func dataSourceRedfishVirtualMediaRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-// 	service, err := NewConfig(m.(*schema.ResourceData), d)
-// 	if err != nil {
-// 		return diag.Errorf(err.Error())
-// 	}
-// 	return readRedfishVirtualMediaCollection(service, d)
-// }
+type DellVirtualMediaDatasource struct {
+	p *redfishProvider
+}
 
-// func readRedfishVirtualMediaCollection(service *gofish.Service, d *schema.ResourceData) diag.Diagnostics {
-// 	var diags diag.Diagnostics
+// Configure implements datasource.DataSourceWithConfigure
+func (g *DellVirtualMediaDatasource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	g.p = req.ProviderData.(*redfishProvider)
+}
 
-// 	//Get manager.Since this provider is thought to work with individual servers, should be only one.
-// 	manager, err := service.Managers()
-// 	if err != nil {
-// 		return diag.Errorf("Error retrieving the managers: %s", err)
-// 	}
+// Metadata implements datasource.DataSource
+func (*DellVirtualMediaDatasource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "virtual_media"
+}
 
-// 	//Get virtual media
-// 	virtualMedia, err := manager[0].VirtualMedia()
-// 	if err != nil {
-// 		return diag.Errorf("Error retrieving the virtual media instances: %s", err)
-// 	}
+// Schema implements datasource.DataSource
+func (*DellVirtualMediaDatasource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "datasource for virtual media.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "ID of the virtual media datasource",
+				Description:         "ID of the virtual media datasource",
+				Computed:            true,
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"redfish_server": schema.ListNestedBlock{
+				MarkdownDescription: "List of server BMCs and their respective user credentials",
+				Description:         "List of server BMCs and their respective user credentials",
+				Validators: []validator.List{
+					listvalidator.SizeAtMost(1),
+					listvalidator.IsRequired(),
+				},
+				NestedObject: schema.NestedBlockObject{
+					Attributes: RedfishServerDatasourceSchema(),
+				},
+			},
+			"virtual_media": schema.ListNestedBlock{
```

Comment:
Why are we making `virtual_media` a block? What is the issue if this was an attribute of type Computed?